strawberryfields/gbs/plot.py

```diff
@@ -0,0 +1,254 @@
+# Copyright 2019 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+r"""
+Plotting and visualization
+==========================
+
+**Module name:** :mod:`strawberryfields.gbs.plot`
+
+.. currentmodule:: strawberryfields.gbs.plot
+
+This module provides functionality for visualizing graphs, subgraphs, and point processes. Graphs
+are plotted using the Kamada-Kawai layout with an aspect ratio of 1:1. The module uses a custom
+Strawberry Fields colour scheme. The standard scheme for graphs uses green nodes and grey edges,
+whereas the scheme for subgraphs uses red nodes and edges.
+
+.. autosummary::
+    plot
+    plot_subgraph
+
+Code details
+^^^^^^^^^^^^
+"""
+from typing import Optional, Tuple
+
+import networkx as nx
+
+
+def _node_coords(graph: nx.Graph, l: dict) -> Tuple:
+    """Converts coordinates for the graph edges for plotting purposes.
+
+    Args:
+        graph (nx.Graph): input graph
+        l (dict[int, float]): dictionary of nodes and their respective coordinates, can be
+        generated using a NetworkX `layout <https://networkx.github.io/documentation/latest/
+        reference/drawing.html#module-networkx.drawing.layout>`__
+
+    Returns:
+         Dict[str, list]: lists of x and y coordinates accessed as keys of a dictionary
+    """
+    n_x = []
+    n_y = []
+
+    for n in graph.nodes():
+        n_x.append(l[n][0])
+        n_y.append(l[n][1])
+
+    return {"x": n_x, "y": n_y}
+
+
+def _edge_coords(graph: nx.Graph, l: dict) -> dict:
+    """Converts coordinates for the graph edges for plotting purposes.
+
+        Args:
+            graph (nx.Graph): input graph
+            l (dict): dictionary of nodes and their respective coordinates
+
+        Returns:
+             dict: x and y coordinates for beginning and end of each edge. `None` is placed as a
+             separator between pairs of nodes/edges.
+        """
+    e_x = []
+    e_y = []
+
+    for e in graph.edges():
+
+        start_x, start_y = l[e[0]]
+        end_x, end_y = l[e[1]]
+
+        e_x.append(start_x)
+        e_x.append(end_x)
+
+        e_y.append(start_y)
+        e_y.append(end_y)
+
+        e_x.append(None)
+        e_y.append(None)
+
+    return {"x": e_x, "y": e_y}
+
+
+GREEN = "#3e9651"
+RED = "#cc2529"
+LIGHT_GREY = "#CDCDCD"
+
+graph_node_colour = GREEN
+graph_edge_colour = LIGHT_GREY
+subgraph_node_colour = RED
+subgraph_edge_colour = RED
+
+graph_node_size = 14
+subgraph_node_size = 16
+
+
```

Comment:
```suggestion

@pytest.mark.skip(reason="Plotting functionality not currently tested")
```
Should we add this? @josh146 
Is this how to turn off testing/codecov reporting for this (and the following) function?
http://doc.pytest.org/en/latest/skipping.html

Reply:
The `pytest.mark.skip` decorator is only used inside the test suite, not the actual codebase.

If you want to skip testing it, all you have to do is not write a test for it 🙂 

If you don't want the coverage to show up either, you can simply add the comment `# pragma: no cover` to the right of the function signature (after the colon).